### PR TITLE
[Fix]count tablet meta's static memory load from disk

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Count segment and tableMeta's static memory which load from disk, and add the memory usage to bvar.

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: be/src/olap/rowset/rowset_meta.cpp [hunk 1]
```diff
@@ -37,10 +37,17 @@
 
 namespace doris {
 
+bvar::Adder<int64_t> g_total_rowset_meta_mem_size("doris_total_rowset_meta_mem_size");
```
Comment (yiguolei) [inline be/src/olap/rowset/rowset_meta.cpp:40]: add a  bvar  g_total_rowset_meta_num

File: be/src/olap/rowset/rowset_meta.cpp [hunk 2]
```diff
@@ -255,6 +264,14 @@ void RowsetMeta::merge_rowset_meta(const RowsetMeta& other) {
     if (rowset_state() == RowsetStatePB::BEGIN_PARTIAL_UPDATE) {
         set_rowset_state(RowsetStatePB::COMMITTED);
     }
+
```
Comment (yiguolei) [inline be/src/olap/rowset/rowset_meta.cpp:267]: add a new method update_meta_size

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: be/src/olap/metadata_adder.h [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    MetadataAdder(const MetadataAdder& other);
+
+public:
+    MetadataAdder();
+
+    void update_metadata_size();
+};
+
+template <typename T>
+MetadataAdder<T>::MetadataAdder(const MetadataAdder<T>& other) {
```
Comment (yiguolei) [inline be/src/olap/metadata_adder.h:76]: 需要重写 operator = ， copy constructor

File: be/src/olap/tablet_schema.h [hunk 4]
```diff
@@ -286,16 +300,18 @@ class TabletIndex {
...
 public:
     enum ColumnType { NORMAL = 0, DROPPED = 1, VARIANT = 2 };
     // TODO(yingchun): better to make constructor as private to avoid
     // manually init members incorrectly, and define a new function like
     // void create_from_pb(const TabletSchemaPB& schema, TabletSchema* tablet_schema).
-    TabletSchema();
-    virtual ~TabletSchema();
+    TabletSchema() = default;
+    virtual ~TabletSchema() = default;
+    TabletSchema(const TabletSchema& other);
```
Comment (yiguolei) [inline be/src/olap/tablet_schema.h:314]: 为什么需要增加这个方法

File: be/src/olap/tablet_schema.h [hunk 5]
```diff
@@ -498,10 +514,14 @@ class TabletSchema {
 
     const std::vector<int32_t>& row_columns_uids() const { return _row_store_column_unique_ids; }
 
+    int64_t get_metadata_size() override;
+
 private:
     friend bool operator==(const TabletSchema& a, const TabletSchema& b);
     friend bool operator!=(const TabletSchema& a, const TabletSchema& b);
 
+    TabletSchema& operator=(const TabletSchema& ts) = default;
```
Comment (yiguolei) [inline be/src/olap/tablet_schema.h:523]: 还有这个方法

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Review by yiguolei — COMMENTED** [review]
std::unique_ptr<ZoneMapIndexReader> _zone_map_index;
    std::unique_ptr<OrdinalIndexReader> _ordinal_index;
    std::unique_ptr<BitmapIndexReader> _bitmap_index;
    std::shared_ptr<InvertedIndexReader> _inverted_index;
    std::shared_ptr<BloomFilterIndexReader> _bloom_filter_index;

Add these counters

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

File: be/src/olap/metadata_adder.h [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    int64_t _current_meta_size {0};
+
+    void add_mem_size(int64_t val);
+
+    void add_num(int64_t val);
+
+protected:
+    virtual ~MetadataAdder();
+
+    virtual int64_t get_metadata_size() { return sizeof(T); }
```
Comment (yiguolei) [inline be/src/olap/metadata_adder.h:114]: change this method to const.

File: be/src/olap/metadata_adder.h [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+
+    MetadataAdder(const MetadataAdder& other);
```
Comment (yiguolei) [inline be/src/olap/metadata_adder.h:116]: move this method before virtual ~MetadataAdder();

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**